lib/src/package.dart

```diff
@@ -500,6 +503,38 @@ Consider removing one of the overrides.
 Cannot override workspace packages.
 
 Package `$override` at `${overriddenWorkspacePackage.presentationDir}` is overridden in `${package.pubspecPath}`.
+''');
+      }
+    }
+  }
+
+  // Check for pubspec.yaml files between the root and any workspace package.
+  final visited = <String>{};
```

Comment:
Just an idea, feel free to ignore.
```suggestion
  final visited = <String>{root.dir};
```
Then you won't need `p.equals(dir, root.dir)`, but then again, maybe this is just too smart. :rofl:

Reply:
I like it!